docs/EN/operate/build.md

```diff
@@ -172,90 +198,136 @@ Sometimes, specific branches are offered for testing. Any desired branch can be
+If a submodule gave an error in response to the `git pull --recurse` command:
 
-* Released (main) branch: Downloads/BuildTrio/Trio_main-[date-time]/Trio
-    * Example: ~/Downloads/BuildTrio/Trio_main-220122-1352/Trio
-* Development (dev) branch: Downloads/BuildTrio/Trio_dev-[date-time]/Trio
-    * Example: ~/Downloads/BuildTrio/Trio_dev-220108-1827/Trio
+* Read which submodule failed to `checkout`
+* Change directory to that module: `cd modulename`
+* Stash your changes: `git stash`
+* Return to the Trio folder and try again: `cd ..; git submodule update`
 
-Use finder to open a Terminal window at the Trio directory by right-clicking on it and selecting `New Terminal at Folder`. 
+If another submodule fails to checkout, repeat the steps for that `modulename`.
 
-Copy each line below and paste it into the terminal window one at a line and hit enter for each line:
+If the modification is one you want to keep, you must restore it:
+ 
+* Change directory to that module: `cd modulename`
+* Restore your changes: `git stash pop`
+* Return to the Trio folder: `cd ..`
+
+The Trio code is now updated.
+
+If Xcode is not open, you can open it by typing `xed .` in the Trio folder of your terminal window.
+
+You can now build the updated Trio app on your phone.
+
+## Xcode Behaviors
+
+You can add Behaviors to your version of Xcode. These custom Behaviors only need to be added once.
+
+You will create a shell script and add it to Xcode.
+
+* `Open Terminal` will open a terminal in the workspace or project folder currently in use by Xcode
+
+### Prepare the Shell Script
+
+Open a new terminal:
+
+Step 1: Create a folder to store your shell scripts by copying and pasting the next line (only do this one time or you will get an error message):
 
 ```
-git stash
+mkdir ~/scripts
 ```
 
+Step 2: Copy and paste this command into your terminal to prepare the `Open Terminal` shell script:
+
 ```
-git fetch
-git pull --recurse
+open -a TextEdit ~/scripts/open_terminal.sh
 ```
 
+Copy and paste the following lines into the editor and then save and close the file.
+
 ```
-git stash pop
+#!/bin/bash
+open -a Terminal "`pwd`"
 ```
 
-If the final `git stash pop` had errors, you must repeat any customizations you made in your prior code—they are no longer compatible with the updated code.
+Step 3: Make the shell script executable; Copy and paste this command into your terminal:
 
-The Trio code is now updated. To open the workspace in Xcode, type `xed .` in your terminal window.
+```
+chmod +x ~/scripts/*.sh
+```
+
+### Add Behaviors to Xcode
+
+Open Xcode
+* Under Xcode menu item, select Behaviors, Edit Behaviors
+* At the bottom of the window, click the `+` sign
+    * Under the Custom section, you should see the New Behavior row, enter `Open Terminal`
+    * On the right side at the bottom, click on Run, then `Choose Script` and select `~/scripts/open_terminal.sh`
```

Comment:
I'm not sure people who aren't skilled in Terminal will know where to find `~/scripts/open_terminal.sh` because they probably don't know what `~` means.

Reply:
added graphics and updated words.